packages/insomnia/src/ui/context/app/insomnia-event-stream-context.tsx

```diff
@@ -75,6 +76,22 @@ interface UserPresenceEvent extends UserPresence {
   type: 'PresentUserLeave' | 'PresentStateChanged' | 'OrganizationChanged' | 'StorageRuleChanged';
 }
 
+const isSameWorkspaceWithRemote = (workspaceId: string | undefined, remoteWorkspaceId: string | undefined) => {
+  if (!workspaceId || !remoteWorkspaceId) {
+    return false;
+  }
+  const vcs = VCSInstance();
+  const currentBackendProject = vcs.getActiveBackendProject();
+  if (
+    currentBackendProject &&
+    currentBackendProject?.id === remoteWorkspaceId &&
+    currentBackendProject.rootDocumentId === workspaceId
+  ) {
+    return true;
+  }
+  return false;
+};
+
```

Comment:
It could be simpler.

```suggestion
const isSameWorkspaceWithRemote = (workspaceId: string | undefined, remoteWorkspaceId: string | undefined) => {
  const vcs = VCSInstance();
  const currentBackendProject = vcs.getActiveBackendProject();
  if (
    workspaceId && remoteWorkspaceId &&
    currentBackendProject?.id === remoteWorkspaceId &&
    currentBackendProject?.rootDocumentId === workspaceId
  ) {
    return true;
  }
  return false;
};

```